internal/exec/terraform_clean.go

```diff
@@ -0,0 +1,432 @@
+			FullPath:     folderPath,
+			RelativePath: relativePath,
+			Files:        []ObjectInfo{},
+		}, nil
+	}
+
+	// Function to collect files for a given folder path
+	collectFilesInFolder := func(folder *Directory, folderPath string) error {
+		for _, pat := range patterns {
+			matchedFiles, err := filepath.Glob(filepath.Join(folderPath, pat))
+			if err != nil {
+				return fmt.Errorf("error matching pattern %s in folder %s: %v", pat, folderPath, err)
+			}
+
+			// Add matched files to folder
+			for _, matchedFile := range matchedFiles {
+				fileInfo, err := addFileInfo(matchedFile)
+				if err != nil {
+					return err
+				}
+				if fileInfo != nil {
+					folder.Files = append(folder.Files, *fileInfo)
+				}
+			}
+		}
+		return nil
+	}
+
+	// Collect files for the base path itself
+	baseFolder, err := createFolder(basePath, filepath.Base(basePath))
+	if err != nil {
+		return nil, err
+	}
+	err = collectFilesInFolder(baseFolder, basePath)
+	if err != nil {
+		return nil, err
+	}
+	if len(baseFolder.Files) > 0 {
+		folders = append(folders, *baseFolder)
+	}
+
+	// Now, search for folders and their files from immediate subdirectories
+	entries, err := os.ReadDir(basePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading the base path %s: %v", basePath, err)
+	}
+
+	for _, entry := range entries {
+		// Only proceed if the entry is a directory
+		if entry.IsDir() {
+			subDirPath := filepath.Join(basePath, entry.Name())
+
+			// Create the folder entry
+			folder, err := createFolder(subDirPath, entry.Name())
+			if err != nil {
+				return nil, err
+			}
+
+			// Collect files in the subdirectory
+			err = collectFilesInFolder(folder, subDirPath)
+			if err != nil {
+				return nil, err
+			}
+
+			// Add folder to the list only if it contains files
+			if len(folder.Files) > 0 {
+				folders = append(folders, *folder)
+			}
+		}
+	}
+
+	return folders, nil
+}
+
+// get stack terraform state files
+func getStackTerraformStateFolder(componentPath string, stack string) ([]Directory, error) {
+	tfStateFolderPath := filepath.Join(componentPath, "terraform.tfstate.d")
+	tfStateFolderNames, err := findFoldersNamesWithPrefix(tfStateFolderPath, stack)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find stack folders: %w", err)
+	}
+	var stackTfStateFolders []Directory
+	for _, folderName := range tfStateFolderNames {
+		tfStateFolderPath := filepath.Join(componentPath, "terraform.tfstate.d", folderName)
+		// Check if exists
+		if _, err := os.Stat(tfStateFolderPath); os.IsNotExist(err) {
+			continue
+		}
+		directories, err := CollectDirectoryObjects(tfStateFolderPath, []string{"*.tfstate", "*.tfstate.backup"})
+		if err != nil {
+			return nil, fmt.Errorf("failed to collect files in %s: %w", tfStateFolderPath, err)
+		}
+		for i := range directories {
+			if directories[i].Files != nil {
+				for j := range directories[i].Files {
+					directories[i].Files[j].Name = folderName + "/" + directories[i].Files[j].Name
```

Comment:
Is this safe to use `/` rather than OS specific character

Reply:
its just added to folder name . I depend on fullpath when delete file